docs/core/diagnostics/event-counter-perf.md

```diff
@@ -0,0 +1,100 @@
+description: Installing and using the dotnet-trace command-line tool.
+ms.date: 07/22/2020
+---
+
+# Tutorial: Measure performance using EventCounters in .NET Core
+
+**This article applies to: ✔️** .NET Core 3.0 SDK and later versions
+
+While an <xref:System.Diagnostics.Tracing.EventSource> is fast, logging too many events for frequent events is still a performance consideration. In this tutorial, you'll learn how an <xref:System.Diagnostics.Tracing.EventCounter> can be used to measure performance of a high frequency of events.
+
+For events that happen every few milliseconds, you'll want the overhead per event to be low (less than a millisecond). Otherwise, the impact on performance will be significant. Logging an event means you're going to write something to disk. If the disk is not fast enough, you will lose events. You need a solution other than logging the event itself.
+
+When dealing with a large number of events, knowing the measure per event is not useful either. Most of the time all you need is just some statistics out of it. So you could crank the statistics within the process itself and then write an event once in a while to report the statistics, that's what <xref:System.Diagnostics.Tracing.EventCounter> will do. Let's take a look at an example how to do this in <xref:System.Diagnostics.Tracing.EventSource?displayProperty=fullName>.
+
+```csharp
+// Give your event sources a descriptive name using the EventSourceAttribute, otherwise the name of the class is used.
+[EventSource(Name = "Samples-EventCounterDemos-Minimal")]
+public sealed class MinimalEventCounterSource : EventSource
+{
+    // Define the singleton instance of the event source
+    public static MinimalEventCounterSource Log = new MinimalEventCounterSource();
+    private EventCounter RequestCounter;
+
+    private MinimalEventCounterSource() : base(EventSourceSettings.EtwSelfDescribingEventFormat)
+    {
+        RequestCounter = new EventCounter("request", this);
+    }
+
+    // Indicate that a request for a URL was made which took a particular amount of time
+    public void Request(string url, float elapsedMSec)
+    {
+        // Notes:
+        //   1. the event ID passed to WriteEvent (1) corresponds to the (implied) event ID
+        //      assigned to this method. The event ID could have been explicitly declared
+        //      using an EventAttribute for this method
+        //   2. Each counter supports a single float value, so conceptually it maps to a single
+        //      measurement in the code.
+        //   3. You don't have to have log with WriteEvent if you don't think you will ever care about details
+        //       of individual requests (that counter data is sufficient).
+
+        // This logs it to the event stream if events are on.
+        WriteEvent(1, url, elapsedMSec);
+        // This adds it to the PerfCounter called 'Request' if PerfCounters are on
+        RequestCounter.WriteMetric(elapsedMSec);
```

Comment:
Assuming we reframe the problem statement as "How do I log some metrics?" instead of "How do I optimize my event collection?" then we could get rid of the WriteEvent(...) portion entirely and the big comment.

Reply:
I opted to leave the `WriteEvent` as it's the only place we show the correlation of writing a metric and event together. All the comments have been removed.